File: src/test/java/recipestore/ResourceLoaderTest.java
package recipestore;

import lombok.SneakyThrows;
import org.hamcrest.Matchers;
import org.junit.Test;
import recipestore.ResourceLoader.Resource;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

import static com.google.common.io.Resources.getResource;
import static java.nio.file.Paths.get;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.jooq.lambda.Seq.of;

public class ResourceLoaderTest {

    @Test
    @SneakyThrows
    public void shouldAccountForAllResourceBundleInClassPath() {
        final Path resourceFolder =
                get(getResource("triplestore.properties").getPath()).getParent();

        final List<String> resourceNames = Files.list(resourceFolder)
                .filter(fileName -> fileName.toString().endsWith(".properties"))
                .map(path -> path.getFileName().toString())
                .collect(Collectors.toList());

        assertThat("ResourceLoader enum should have all the resources ",
                of(Resource.values()).map(Object::toString).map(name -> name + ".properties").toList(), Matchers.containsInAnyOrder(resourceNames.toArray()));

    }

    @Test
    public void shouldGetAvailableResource() throws Exception {

        final Optional<String> useEmbedded = ResourceLoader.get.apply(Resource.triplestore, "useEmbedded");
        assertThat(useEmbedded.isPresent(), Matchers.equalTo(true));

    }

    @Test
    public void shouldReturnEmptyForUnavailableResource() throws Exception {

        final Optional<String> useEmbedded = ResourceLoader.get.apply(Resource.triplestore, "somethineElse");
        assertThat(useEmbedded.isPresent(), Matchers.equalTo(false));

    }
}

File: src/test/scala/recipestore/ResourceLoaderTest.scala
package recipestore

import java.nio.file.Paths.get
import java.nio.file.{Files, Path}

import com.google.common.io.Resources.getResource
import org.scalatest.{FunSuite, Matchers}

import scala.collection.JavaConverters._

class ResourceLoaderTest extends FunSuite with Matchers {
  test("Should account for all resources in the path") {
    val resourceFolder: Path = get(getResource("triplestore.properties").getPath).getParent
    val resourceNames: List[String] = Files.list(resourceFolder)
      .iterator().asScala.toList
      .filter(p => p.getFileName.endsWith("properties"))
      .map(p => p.getFileName.toString)
    val actual = AppResource.values
      .map(_.toString)
      .map(name => name + ".properties")
      .toList
    actual should be equals (resourceNames)
  }

  test("Should get a valid resource") {
    val useEmbedded: Option[String] = ResourceLoader(AppResource.TriplestoreResource, "useEmbedded")
    useEmbedded.isDefined should be equals (true)
  }

  test("Shoudl get null value for invalid resource") {
    val useEmbedded: Option[String] = ResourceLoader(AppResource.TriplestoreResource, "somethineElse")
    useEmbedded.isDefined should be equals (false)
  }
}